### PR TITLE
Extact_code can detect single-line code now

### PR DESCRIPTION
**Pull request by BeibinLi**
## Why are these changes needed?

A parameter is added. Default to False.
When True, it will also detect single-line code.

For instance, 
data = """

hello, how are you.

You can run `python main.py`

Or, you can write
```python
1 + 1 == 2
print(3)
```

```
Formated stuff here
and there
```

Otherwise, do `source download.sh` for it.

I would also want to extract variable `x`.

Is that ok?
"""

extract_code(data, detect_single_line_code=True)

We will get 
```
[('UNKNOWN', 'python main.py'),
 ('python', '1 + 1 == 2\nprint(3)'),
 ('UNKNOWN', 'Formated stuff here\nand there'),
 ('UNKNOWN', 'source download.sh'),
 ('UNKNOWN', 'x')]
```

## Checks

- [X] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [X] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by sonichi** [comment]
Could you add test to cover the new code?

File: autogen/code_utils.py [hunk 1]
```diff
@@ -34,24 +34,38 @@ def infer_lang(code):
...
-    # If a match is found, return the code
-    # if match:
-    #     return match.group(2), match.group(1)
-    # If no code block is found, return the whole text
-    return match if match else [(UNKNOWN, text)]
+    if not detect_single_line_code:
+        match = re.findall(pattern, text, flags=re.DOTALL)
+        return match if match else [(UNKNOWN, text)]
+
+    code_pattern = re.compile(r"`{3}(\w+)?\s*([\s\S]*?)`{3}|`([^`]+)`")
```
Comment (pcdeadeasy) [inline autogen/code_utils.py:56]: Please add a comment to explain the intent of the regex

**Review by pcdeadeasy — COMMENTED** [review]
Please address the suggestion to add the comment

**Comment by BeibinLi** [comment]
@sonichi  @pcdeadeasy 

1. Comments added.
2. Test case added/modified.

File: autogen/code_utils.py [hunk 2]
```diff
@@ -34,24 +34,43 @@ def infer_lang(code):
...
-        pattern (Optional, str): The regular expression pattern for finding the code block.
+        pattern (str, optional): The regular expression pattern for finding the
+            code block. Defaults to CODE_BLOCK_PATTERN.
+        detect_single_line_code (bool, optional): Enable the new feature for
+            extracting single line code. Defaults to False.
 
     Returns:
         list: A list of tuples, each containing the language and the code.
+          If there is no code in the input text, the language would be "unknown".
+          If there is code but the language is not specified, the language would be "".
```
Comment (sonichi) [inline autogen/code_utils.py:52]:
```suggestion
          If there is no code block in the input text, the language would be "unknown".
          If there is code block but the language is not specified, the language would be "".
```

**Review by pcdeadeasy — APPROVED** [review]
Looks good to me.

**Pull request merged**